custom/logistics/api.py

```diff
@@ -251,6 +251,11 @@ def web_user_sync(self, ilsgateway_webuser):
             if self.domain not in user.get_domains():
                 user.add_domain_membership(self.domain, location_id=location_id)
                 user.save()
+            else:
+                dm = user.get_domain_membership(self.domain)
+                if dm.location_id != location_id:
+                    dm.location_id = location_id
```

Comment:
do we need to check for role id as well? (i think you may have already done so actually in the other PR i merged pertaining to web user sync)